### PR TITLE
[SPARK-26033][SPARK-26034][PYTHON][FOLLOW-UP] Small cleanup and deduplication in ml/mllib tests

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR is a small follow up that puts some logic and functions into smaller scope and make it localized, and deduplicate.

## How was this patch tested?

Manually tested. Jenkins tests as well.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @BryanCutler - it's a small minor followup.

**Comment by HyukjinKwon** [comment]
retest this please

**Review by srowen — APPROVED** [review]

File: python/pyspark/mllib/tests/test_linalg.py [hunk 1]
```diff
@@ -22,33 +22,18 @@
...
-ser = make_serializer()
-
-
-def _squared_distance(a, b):
-    if isinstance(a, Vector):
-        return a.squared_distance(b)
-    else:
-        return b.squared_distance(a)
+from pyspark.testing.mllibutils import MLlibTestCase
+from pyspark.testing.utils import have_scipy
```
Comment (BryanCutler) [inline python/pyspark/mllib/tests/test_linalg.py]: Oh that's good, didn't realize have_scipy was there

**Review by BryanCutler — APPROVED** [review]
LGTM

**Comment by BryanCutler** [comment]
merged to master, thanks @HyukjinKwon

**Comment by HyukjinKwon** [comment]
Thank you @srowen and @BryanCutler

**Pull request closed without merging**